numbers_server.py

```diff
@@ -162,55 +178,43 @@ def start_server():
                 if active_socket == server_socket:
-                    client_socket, client_address = server_socket.accept()
-                    sockets_list.append(client_socket)
-                    clients[client_socket] = client_address
-                    print_login_message(client_socket)
+                    accept_socket(server_socket)
 
                 # Step 10: Otherwise, it's an existing client socket with data
                 else:
                     # gets the address of the active socket for dictionary searches
                     client_address = clients[active_socket]
 
+                    #recive packet
                     if client_address not in clients_messages:
                         clients_messages[client_address]=""
                     clients_messages[client_address] += active_socket.recv(1024)
-
                     message_builder = clients_messages[client_address].decode("utf-8")
 
                     # check if we got the full message
-                    if message_builder[-1] == "$":
+                    if message_builder[-1] == "\t":
                         # Step 10.5 if you're not logged in, do so. if you don't i kick u
                         if active_socket not in logged_in_users_sockets_list:
-
-                            username, password = process_login_data(message_builder)
-
-                            if username in known_users_dict and known_users_dict[username] == password:
-                                logged_in_users_sockets_list.append(active_socket)
-                            else:
-                                # the user can try again so no need to close the socket
-                                active_socket.send("Failed to login.$".encode("utf-8"))
+                            handle_new_user(message_builder, active_socket, known_users_dict, logged_in_users_sockets_list)
                             continue
 
                         # Step 13: you're logged in, so you probably sent us some command
                         else:
                             # handle this case first because it requires removing stuff from data structures
-                            if message_builder == 'quit$':
-                                send_invalid_input_error(active_socket)
-                                clients_messages.pop(client_address)
-                                logged_in_users_sockets_list.remove(active_socket)
-                                clients.pop(active_socket)
-                                disconnect_socket(active_socket)
-
-                            message_to_send = command_handler(active_socket, message_builder)+'$'
+                            if message_builder == 'quit\t':
+                                close_connection(active_socket, client_address,logged_in_users_sockets_list)
+                                continue
+
+                            message_to_send = command_handler(active_socket, message_builder)+'\t'
```

Comment:
i would suggest not sending it with a \t since we are parsing it. 
IMO it makes more sense to send it without a +'\t'  and just expect commands as is, and when we parse a msg just drop the \t